superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FilterScope/FilterScope.tsx

```diff
@@ -78,10 +80,13 @@ const FilterScope: FC<FilterScopeProps> = ({
 
   const onUpdateFormValues = useCallback(
     (formValues: any) => {
+      if (formScopingType === ScopingType.specific) {
+        setLastSpecificScope(formValues.scope);
```

Comment:
I personally prefer to use `useRef` for this case (since it won't need to re-render)

Reply:
Good call! Done.